### PR TITLE
fix(imports) possible fix for imports running out of memory

### LINKED ISSUE #29162: Slow performance with imports in some cases
### Parent Issue

_No response_

### Problem Statement

We have seen some cases where importing content can take a long time.  Additionally, when importing content, it spins up a new thread, which does not report to glowroot, making identifying these issues difficult.  We need to optimize for this.

### Steps to Reproduce

Create a content type with 20 fields.  Run a csv import of a large amount of records for that type with glowroot attached to the dotCMS

There will be no glowroot reporting/traces to help identify slow code.

### Acceptance Criteria

We can add some blind fixes here but the real test will be if we can get the import to work nicely in an environment that is sized down.

### dotCMS Version

24.05

### Proposed Objective

Application Performance

### Proposed Priority

Priority 2 - Important

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by wezell**
This does a few things:
- Removes reading the file in as a byte[] and instead passes around a `File` object
- Changes how the import is run inline (helpful for glowroot traces) or to use the DotSubmitter thread pool.  Config var is `IMPORT_CONTENTLETS_ASYNC`
- Commit Granularity changed to 100, meaning every 100 contentlets will be committed and a new transaction will be started.
- Copied CSV file to tmp dir.

ref: #29162

This PR fixes: #29162

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

**Review by erickgonzalez — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ImportContentletsAction.java [hunk 1]
```diff
@@ -138,19 +137,14 @@ else if (bytes == null || bytes.length == 0) {
 						return;
 					}
 
-					try {
-						Reader reader = null;
-						CsvReader csvreader = null;
+					Charset charset = importContentletsForm.getLanguage() == -1 ? Charset.defaultCharset() : FileUtil.detectEncodeType(file);
+					try(Reader reader = new BufferedReader(new InputStreamReader(new FileInputStream(file), charset))){
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ImportContentletsAction.java:141]: should we use Files.newInputStream

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ImportContentletsAction.java [hunk 2]
```diff
@@ -304,7 +291,15 @@ else if(eCode != null) {
 						}
 					}
 				};
-				 t.start();
+
+
+				if(Config.getBooleanProperty("IMPORT_CONTENTLETS_ASYNC", true)) {
+					DotConcurrentFactory.getInstance().getSubmitter("importContentlets").submit(runnable);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ImportContentletsAction.java:297]: Nice

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ImportContentletsAction.java (hunk 1 above)
Comment (wezell) [reply dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/action/ImportContentletsAction.java:141]: yeah, totally.

**Review by wezell — COMMENTED** [review]
Awesome work Erick

**Pull request merged**